raiden/network/transport/matrix/client.py

```diff
@@ -165,6 +165,29 @@ def _send(self, method, path, *args, **kwargs):
         else:
             raise last_ex
 
+    def send_to_device(self, event_type, messages, txn_id=None):
+        started = time.time()
+        last_ex = None
+        for delay in self.retry_delay():
+            try:
+                with self._semaphore:
+                    return super().send_to_device(event_type, messages, txn_id=None)
+            except (MatrixRequestError, MatrixHttpLibError) as ex:
+                # from MatrixRequestError, retry only 5xx http errors
+                if isinstance(ex, MatrixRequestError) and ex.code < 500:
+                    raise
+                if time.time() > started + self.retry_timeout:
+                    raise
+                last_ex = ex
+                log.debug(
+                    'Got http _send exception, waiting then retrying',
+                    wait_for=delay,
+                    _exception=ex,
+                )
+                gevent.sleep(delay)
```

Comment:
Please, on Runnables, avoid as much as possible `gevent.sleep`, it breaks proper `stop`. Use `_stop_event.wait` instead, with proper checks if it bailed out early because the event was set and break the loop early.